include/exadg/acoustic_conservation_equations/spatial_discretization/operators/grad_div_operator.h

```diff
@@ -0,0 +1,537 @@
+      {
+        vector const flux_momentum = kernel.get_volume_flux_weak_momentum(velocity, q);
+        scalar const flux_mass     = kernel.get_volume_flux_weak_mass(pressure, q);
+
+        pressure.submit_gradient(rhocc * flux_momentum, q);
+        velocity.submit_divergence(rho_inv * flux_mass, q);
+      }
+      else if constexpr(formulation == FormulationGradientDivergenceTerms::Strong)
+      {
+        scalar const flux_momentum = kernel.get_volume_flux_strong_momentum(velocity, q);
+        vector const flux_mass     = kernel.get_volume_flux_strong_mass(pressure, q);
+
+        pressure.submit_value(rhocc * flux_momentum, q);
+        velocity.submit_value(rho_inv * flux_mass, q);
+      }
+      else if constexpr(formulation == FormulationGradientDivergenceTerms::SkewSymmetric)
+      {
+        vector const flux_momentum = kernel.get_volume_flux_weak_momentum(velocity, q);
+        vector const flux_mass     = kernel.get_volume_flux_strong_mass(pressure, q);
+
+        pressure.submit_gradient(rhocc * flux_momentum, q);
+        velocity.submit_value(rho_inv * flux_mass, q);
+      }
+      else
+      {
+        AssertThrow(false, dealii::ExcMessage("Not implemented."));
+      }
+    }
+  }
+
+  template<FormulationGradientDivergenceTerms formulation,
+           bool                               consider_exterior_cell,
+           typename ExteriorFaceIntegratorP,
+           typename ExteriorFaceIntegratorU>
+  void
+  do_face_integral(FaceIntegratorP &         pressure_m,
+                   ExteriorFaceIntegratorP & pressure_p,
+                   FaceIntegratorU &         velocity_m,
+                   ExteriorFaceIntegratorU & velocity_p,
+                   Number const              rho_inv,
+                   Number const              tau,
+                   Number const              rhocc,
+                   Number const              gamma) const
+  {
+    for(unsigned int q : pressure_m.quadrature_point_indices())
+    {
+      scalar const pm = pressure_m.get_value(q);
+      scalar const pp = pressure_p.get_value(q);
+      vector const um = velocity_m.get_value(q);
+      vector const up = velocity_p.get_value(q);
+      vector const n  = pressure_m.normal_vector(q);
+
+      vector const flux_momentum =
+        kernel.calculate_lax_friedrichs_flux_momentum(um, up, gamma, pm, pp, n);
+
+      scalar const flux_mass = kernel.calculate_lax_friedrichs_flux_mass(pm, pp, tau, um, up, n);
+
+      if constexpr(formulation == FormulationGradientDivergenceTerms::Weak)
+      {
+        scalar const flux_momentum_weak = rhocc * flux_momentum * n;
+        vector const flux_mass_weak     = rho_inv * flux_mass * n;
+
+        pressure_m.submit_value(flux_momentum_weak, q);
+        velocity_m.submit_value(flux_mass_weak, q);
+
+        if constexpr(consider_exterior_cell)
+        {
+          // minus sign since n⁺ = - n⁻
+          pressure_p.submit_value(-flux_momentum_weak, q);
```

Comment:
From this line, I expected to find a function definition of `BoundaryFaceIntegratorP::submit_value()`, but I did not find it. 

For this reason, I actually do not like the present design very much. It seems to be too implicit in my opinion.

Reply:
As discussed in person, we keep this for now and see how other parts of ExaDG could be realized this way in the future.